proparse/src/main/java/org/prorefactor/proparse/antlr4/Lexer.java

```diff
@@ -0,0 +1,1117 @@
+        case '\t':
+        case '\n':
+        case '\f':
+        case '\r':
+        case ' ':
+          getChar();
+          return whitespace();
+
+        case '"':
+        case '\'':
+          if (prepro.isEscapeCurrent()) {
+            getChar();
+            // Escaped quote does not start a string
+            return id(PreprocessorParser.FILENAME);
+          } else {
+            currStringType = currInt;
+            getChar();
+            return quotedString();
+          }
+
+        case '/':
+          getChar();
+          if (currChar == '*') {
+            return comment();
+          } else if (currChar == '/') {
+            return singleLineComment();
+          } else if (currChar == '(' || currIsSpace()) {
+            // slash (division) can only be followed by whitespace or '('
+            // ...that's what I found empirically, anyway. (jag 2003/05/09)
+            return makeToken(PreprocessorParser.SLASH);
+          } else {
+            append();
+            getChar();
+            return id(PreprocessorParser.FILENAME);
+          }
+
+        case ':':
+          getChar();
+          return colon();
+
+        case '&':
+          getChar();
+          return ampText();
+        case '@':
+          getChar();
+          if (currIsSpace())
+            return makeToken(PreprocessorParser.LEXAT);
+          else
+            append();
+          getChar();
+          return id(PreprocessorParser.ANNOTATION);
+        case '[':
+          getChar();
+          return makeToken(PreprocessorParser.LEFTBRACE);
+        case ']':
+          getChar();
+          return makeToken(PreprocessorParser.RIGHTBRACE);
+        case '^':
+          getChar();
+          return makeToken(PreprocessorParser.CARET);
+        case ',':
+          getChar();
+          return makeToken(PreprocessorParser.COMMA);
+        case '!':
+          getChar();
+          return makeToken(PreprocessorParser.EXCLAMATION);
+        case '=':
+          getChar();
+          return makeToken(PreprocessorParser.EQUAL);
+        case '(':
+          getChar();
+          return makeToken(PreprocessorParser.LEFTPAREN);
+        case ')':
+          getChar();
+          return makeToken(PreprocessorParser.RIGHTPAREN);
+        case ';':
+          getChar();
+          return makeToken(PreprocessorParser.SEMI);
+        case '*':
+          getChar();
+          return makeToken(PreprocessorParser.STAR);
+        case '?':
+          getChar();
+          return makeToken(PreprocessorParser.UNKNOWNVALUE);
+        case '`':
+          getChar();
+          return makeToken(PreprocessorParser.BACKTICK);
+
+        case '0':
+        case '1':
+        case '2':
+        case '3':
+        case '4':
+        case '5':
+        case '6':
+        case '7':
+        case '8':
+        case '9':
+          getChar();
+          return digitStart();
+
+        case '.':
+          getChar();
+          return periodStart();
+
+        case '>':
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 8 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1151)